AsyncDisplayKit/ASVideoNode.mm

```diff
@@ -168,9 +198,54 @@ - (void)didLoad
         dispatch_async(dispatch_get_main_queue(), ^{
           _placeholderImageNode.frame = self.bounds;
           [self insertSubnode:_placeholderImageNode atIndex:0];
+          // [self setNeedsLayout];
         });
-      }];
-    });
+      }
+    }];
+  });
+}
+
+- (void)interfaceStateDidChange:(ASInterfaceState)newState fromState:(ASInterfaceState)oldState
+{
+  if (!(newState & ASInterfaceStateVisible)) {
```

Comment:
For readability, make two local BOOLs - nowVisible and wasVisible.

Not sure if they link properly in this file as there are some weird problems with them, but see if you can use ASInterfaceStateIncludesVisible()